### PR TITLE
feat(examples): nft marketplaceft calimero update

### PR DESCRIPTION
**Pull request by kuzmicivan**
## Summary:
NFT Marketplace example working on Calimero Network private shard.

## Test plan:

https://user-images.githubusercontent.com/105968375/229251530-feed6bc0-dfa0-4161-b561-d3ff5e678a2f.mov

https://user-images.githubusercontent.com/105968375/229251567-ba9d15b9-d43a-4052-9451-ba389a390987.mov

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have added a test plan that prove my fix is effective or that my feature works
- [x] I squashed all commits and provided a meaningful commit message
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: nft-market/pages/mint.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      title={translations.home.title}
+      nearLogin={login}
+      nearLogout={() => {
+        logout();
+        router.reload();
+      }}
+      nearSignedIn={isSignedIn}
+      openMint={true}
+    >
+      <div className="flex justify-center text-white">
```
Comment (frdomovic) [inline nft-market/pages/mint.tsx]: this whole section needs to be extracted to component

File: nft-market/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    <PageWrapper
+      isSignedIn={isSignedIn}
+      title={translations.home.title}
+      nearLogin={login}
+      nearLogout={() => {
+        logout();
+        router.reload();
+      }}
+      nearSignedIn={isSignedIn}
+    ></PageWrapper>
```
Comment (frdomovic) [inline nft-market/pages/index.tsx]: What will index page display?

File: nft-market/pages/buynfts.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      nearLogin={login}
+      nearLogout={() => {
+        logout();
+        router.reload();
+      }}
+      nearSignedIn={isSignedIn}
+      openBuy={true}
+    >
+      <div className="w-full h-scree grid grid-cols-3 gap-8">
+        {nftsForSell.map((nft) => {
```
Comment (frdomovic) [inline nft-market/pages/buynfts.tsx]: all this logic should be moved to new component called NFTContainer or something like that

File: nft-market/package copy.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (frdomovic) [inline nft-market/package copy.json]: do we need this?

File: nft-market/next.config.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  reactStrictMode: true,
+  images: {
+    domains: ['*', 'picsum.photos', 'images.unsplash.com']
```
Comment (frdomovic) [inline nft-market/next.config.js]: just leave as '*'

**Review by frdomovic — CHANGES_REQUESTED** [review]
Why is this being merged into master and not other PR?

File: nft-market/pages/mint.tsx (hunk 1 above)
Comment (kuzmicivan) [reply nft-market/pages/mint.tsx]: Made a mintDialog component

File: nft-market/pages/index.tsx (hunk 2 above)
Comment (kuzmicivan) [reply nft-market/pages/index.tsx]: Added a some home page design. Check it out and give me feedback if it is good for now.

File: nft-market/pages/buynfts.tsx (hunk 3 above)
Comment (kuzmicivan) [reply nft-market/pages/buynfts.tsx]: Made a nftsContainer component

File: nft-market/package copy.json (hunk 4 above)
Comment (kuzmicivan) [reply nft-market/package copy.json]: No we don't. Removed that file

File: nft-market/next.config.js (hunk 5 above)
Comment (kuzmicivan) [reply nft-market/next.config.js]: Did your suggestion

File: nft-market/ui/components/cantBuyYourOwnNftDialog/CantBuyYourOwnNftDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { PopupContainer } from "../popContainer/PopupContainer";
+import translations from "../../constants/en.global.json";
+import Close from "../closeButton/CloseButton";
+
+interface CantBuyYourOwnNftDialogProps {
+  onClose: () => void | undefined;
```
Comment (frdomovic) [inline nft-market/ui/components/cantBuyYourOwnNftDialog/CantBuyYourOwnNftDialog.tsx]: why void or undefined?

File: nft-market/ui/components/mintDialog/MintDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    setErrorDialog(false);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const regex = /^(ftp|http|https):\/\/[^ "]+$/;
+    const { title, description, media } = nftForm;
+    !media
+      ? setUnvalidUrl("URL cannot be blank!")
+      : !regex.test(media) && setUnvalidUrl("Provided URL is invalid!");
```
Comment (frdomovic) [inline nft-market/ui/components/mintDialog/MintDialog.tsx]: you can add this to utils folder as test function with its .test.ts file

**Review by frdomovic — APPROVED** [review]
Really good, good job! Check comments if interface can be updated like:  onClose?: () => void; if not good to merge

File: nft-market/ui/components/cantBuyYourOwnNftDialog/CantBuyYourOwnNftDialog.tsx (hunk 6 above)
Comment (kuzmicivan) [reply nft-market/ui/components/cantBuyYourOwnNftDialog/CantBuyYourOwnNftDialog.tsx]: Removed it. It was unnecessary.

**Pull request merged**